base/io.jl

```diff
@@ -1520,3 +1520,17 @@ function countlines(io::IO; eol::AbstractChar='\n')
 end
 
 countlines(f::AbstractString; eol::AbstractChar = '\n') = open(io->countlines(io, eol = eol), f)::Int
+
+# From Pkg.jl
```

Comment:
Would it be easier to make Pkg instead use an IOContext{IO}(io), since it has the same implementation as this, without needing to add code to Base just for Pkg?

Reply:
I was thinking about that, but `IOContext` has a type parameter indicating the parent type. I thought the objective was to create an abstract inference barrier so as to not force recompilation of Pkg methods for new `IO` types.

I'm thinking of splitting the pull request so that we can focus on this issue.

Reply:
That is true of the usual interface, but it is valid for the parameter to be simply `IO` also. Someone could make a PR to make that constructor a bit simpler also, as right now it is potentially a bit awkwardly specified as `IOContext{IO}(unwrapcontext(io)...)` and does not reliably propagate when making a new IOContext from an existing one with more keys.

Reply:
I see so I may be able to define the following.

```julia
UnstableIO(io:IO) = IOContext{IO}(inferencebarrier(io))
```